docs/querying/functions.md

```diff
@@ -464,6 +464,89 @@ by the number of seconds under the specified time range window, and should be
+identifies an info metric. The remaining info metric labels are considered
+"data" labels (i.e. non-identifying). The purpose of an info metric's 
+identifying labels is to connect it to regular (non-info) metrics. It's similar
+to foreign keys in SQL, that allow you to combine data spread across multiple
+tables. The data labels in practice encode metadata key value pairs, and are
+what one wants to add to query results (identifying labels are already 
+included).
+
+Before the `info` function, the solution to including info metric data labels
+in Prometheus queries has been to so-called "join" with the relevant info metric.
+Take the following example:
+
+```
+rate(http_server_request_duration_seconds_count[2m])
+* on (job, instance) group_left (k8s_cluster_name)
+target_info
+```
+
+The core of the query is the expression `rate(http_server_request_duration_seconds_count[2m])`.
+However, the resulting time series are joined with `target_info` (an info 
+metric) series that have the same `job` and `instance` labels (these are 
+identifying). When joining, the `k8s_cluster_name` label is picked from 
+`target_info` and added to the time series.
+
+Writing queries like the above is rather tricky, and also doesn't insulate the
+user from possible "join conflicts". Join conflicts happen temporarily when
+non-identifying labels of an info metric change, without the old version of the
+metric getting marked as stale. In order to provide a better user experience,
+we have developed the `info` function. Using `info`, the preceding example can
+instead be written like so:
+
+```
+info(
+  rate(http_server_request_duration_seconds_count[2m]),
+  {k8s_cluster_name=~".+"}
+)
+```
+
+By wrapping the core expression (`rate(http_server_request_duration_seconds_count[2m])`)
+in an `info` call, combined with the data label matcher `k8s_cluster_name=~".+"`,
+we also achieve the effect of joining the resulting time series with 
+`target_info` on the `job` and `instance` labels, and picking the 
+`k8s_cluster_name` label from the latter.
+
+If the second argument to `info` (`{k8s_cluster_name=~".+"}`) is omitted, all
+of `target_info`'s data labels are instead picked.
+
+When using `info`, join conflicts are avoided by automaticaly picking the 
+newest series.
+
+While `info` normally automatically finds info metrics, it's possible to
+restrict them by providing a `__name__` label matcher, e.g.
+`{__name__="target_info"}`.
+
+### Limitations
+
+In its current iteration, `info` defaults to considering only one info metric:
+`target_info`. It also assumes that the identifying info metric labels are
+`instance` and `job`. `info` does support other info metrics however, through
+`__name__` label matchers. E.g., one can explicitly say to consider both 
+`target_info` and `build_info` as follows:
+`{__name__=~"(target|build)_info"}`. However, the identifying labels always
+have to be `instance` and `job`.
```

Comment:
So far, we have been switching between "info series" and "info metrics". Maybe we should consistently use "info series". I have tried to do that in my suggestions above, so maybe let's do it here, too.

Reply:
I don't think consistently using "info series" always makes sense, since "info metric" refers to the "family" of series with the same name name, e.g. `target_info`. E.g. the following wording wouldn't be true, since there can be an indefinite number of `target_info` series:

> In its current iteration, `info` defaults to considering only one info series:

Reply:
Giving it some more thought, I could reword around the "series" term.

Reply:
PTAL.